lib/migrate/extractTextPlugin/extractTextPlugin.js

```diff
@@ -1,58 +1,145 @@
+	 * @param {Function} action - called with sanitized path
+	 * @returns {*} - null
+	 */
+	function pathSwitch(path, action) {
+		if (isType(path, "ArrayExpression")) {
+			action(path.elements);
+			return;
 		}
-		return p;
-	};
-	const name = utils.findVariableToPlugin(
-		j,
-		ast,
-		"extract-text-webpack-plugin"
-	);
-	if (!name) return ast;
-
-	return ast
-		.find(j.CallExpression)
-		.filter(p => findInvocation(j, p, name))
-		.forEach(changeArguments);
+		action([path]);
+	}
+
+	const filename = getLoaderOption(
+		"filename",
+		["parent", "value", "value", "value"],
+		["parent", "parent", "parent", "value", "callee"]);
+
+	const publicPath = getLoaderOption(
+		"publicPath",
+		["parent", "value", "value", "value"],
+		["parent", "parent", "parent", "value", "callee", "object"]);
+
+	const currentFallbackLoader = getLoaderOption(
+		"fallback",
+		["parent", "value", "value"],
+		["parent", "parent", "parent", "value", "callee", "object"]);
+
+	const currentLoader = getLoaderOption(
+		"use",
+		["parent", "value", "value"],
+		["parent", "parent", "parent", "value", "callee", "object"]);
+
+	// Remove require for old plugin
+	ast.find(j.Identifier, { name: "ExtractTextPlugin" })
+		.filter(path => safeTraverse(path, ["parent", "parent", "value"]))
+		.filter(path => isType(path.parent.parent.value, "VariableDeclaration"))
+		.forEach(path => {
+			j(path.parent.parent).remove();
+			requireExists = true;
+		});
+
+	// Add require for new plugin
+	const pathRequire = getRequire(j, "MiniCssExtractPlugin", "mini-css-extract-plugin");
+	requireExists && ast.find(j.Program).replaceWith(p => j.program([].concat(pathRequire).concat(p.value.body)));
+
+	// Remove old plugin
+	findPluginsByName(j, ast, ["ExtractTextPlugin"])
+		.filter(path => safeTraverse(path, ["parent", "value"]))
+		.forEach(path => {
+			j(path).remove();
+			pluginExists = true;
+		});
+
+	// Add new plugin
+	pluginExists && ast.find(j.Identifier, { name: "plugins" })
+		.filter(path => safeTraverse(path, ["parent", "value"]))
+		.forEach(path => {
+			createOrUpdatePluginByName(j, path.parent.value, "MiniCssExtractPlugin", { filename, publicPath });
+		});
+
+	// Remove old loader
+	ast.find(j.Identifier, { name: "ExtractTextPlugin" })
+		.filter(path => safeTraverse(path, ["parent", "value"]))
+		.filter(path => isType(path.parent.value, "MemberExpression"))
+		.forEach(path => {
+			loaderPath = safeTraverse(path, ["parent", "parent", "parent", "value"]);
+			j(path.parent.parent).remove();
+		});
+
+	// Build loader arrays
+	currentLoader && pathSwitch(currentLoader, switchedLoader => {
+		prodLoaders.push(loader, ...switchedLoader);
+		currentFallbackLoader && pathSwitch(currentFallbackLoader, switchedFallbackLoader => {
+			devLoaders.push(...switchedFallbackLoader, ...switchedLoader);
+		});
+	});
+
+	// Add new loader
+	loaderPath && Object.assign(loaderPath, {
+		value: currentFallbackLoader ?
+			j.conditionalExpression(
```

Comment:
`MiniCssExtractPlugin` does not support `fallback` and `disabled` options. Instead, users should manually swap out `MiniCssExtractPlugin.loader` with the loader that was normally used as a `fallback`, which is what we're doing with this ternary. More information can be found [here](https://github.com/webpack-contrib/mini-css-extract-plugin/issues/88#issuecomment-3786856530).